apisupport/maven.apisupport/src/org/netbeans/modules/maven/apisupport/NbmWizardIterator.java

```diff
@@ -145,11 +149,11 @@ public Set<FileObject> instantiate() throws IOException {
             Map<String,String> additional = Collections.singletonMap("netbeansVersion", version); // NOI18N
             
             if (archetype == NB_MODULE_ARCH) {
-                getLatestArchetypeVersion(NB_MODULE_ARCH);
+                updateToLatestKnownArchetypeVersion(NB_MODULE_ARCH);
                 NBMNativeMWI.instantiate(vi, projFile, version, Boolean.TRUE.equals(wiz.getProperty(OSGIDEPENDENCIES)), null);
 
             } else {
-                getLatestArchetypeVersion(NB_APP_ARCH);
+                updateToLatestKnownArchetypeVersion(NB_APP_ARCH);
                 ArchetypeWizards.createFromArchetype(projFile, vi, archetype, additional, true);
```

Comment:
i noticed that the NBM wizard isn't even using the archetype. It builds the pom manually, the NB-APP wizard uses the archetype.

wondering why there is this destination

Reply:
The archetype were at Mojohaus, created by Milos Kleint and Jesse Glick at the time they were at Oracle. Same was from nbm-maven-plugin. Milos and Jesse were doing release and also accepted the move to Apache. But the archetype didn't recieve any PR from 2015 until donation. 
We have the hands on archetype and netbeans so we can do what we want now.

Reply:
ah ok. So lets keep it as is for now, but we should probably change it so that both wizards use the archetypes (via `ArchetypeWizards.createFromArchetype()`). Right now its two entirely different code paths.

Reply:
just I was not an insider it was it look like from external perspective.